### PR TITLE
Allow `FormData` to be passed as a `body` to `Fetcher`

### PR DESCRIPTION
**Pull request by trevor-scheer**
Add a `TFormData` generic type which allows `Fetcher` consumers to extend the `body` union with additional types (with the intention that the type provided is for a `FormData` type.

### CODE REVIEW

File: packages/fetcher/src/__tests__/index.test.ts [hunk 1]
```diff
@@ -1,19 +1,130 @@
...
+      body: formData,
+    });
+
+    const t = await result.text();
+    expect(t).toMatchInlineSnapshot(`"ok"`);
+  });
 });
+
+// Ensures an active and clean nock before every test
+function nockBeforeEach() {
```
Comment (glasser) [inline packages/fetcher/src/__tests__/index.test.ts:128]: i thought you added these somewhere in a different PR?

**Review by glasser — APPROVED** [review]
This looks good. If we can avoid actually publishing to npm before we verify that we can make it work as we'd like with RESTDataSource then that would be best probably. (Or publish a prerelease or whatever.)

**Comment by glasser** [comment]
For context, this is to allow folks who use RESTDataSource to use FormData (we don't have any need to use it in Apollo Server/Gateway/etc).

File: packages/fetcher/src/__tests__/index.test.ts (hunk 1 above)
Comment (trevor-scheer) [reply packages/fetcher/src/__tests__/index.test.ts:128]: Maybe just time to extract them into a util package? They don't exist in this repo yet. You're thinking of the usage reporting PR.

**Pull request merged**